spec/integration/client/client_spec.rb

```diff
@@ -31,8 +32,48 @@ def stop_tiny_server
     @server = @api = nil
   end
 
+  def install_certificate_in_store
+    if ChefUtils.windows?
+      powershell_exec!("New-SelfSignedCertificate -certstorelocation cert:\\localmachine\\my -Subject #{client_name} -FriendlyName #{client_name} -KeyExportPolicy Exportable")
+    end
+  end
+
+  def create_registry_key
+    powershell_exec! <<~EOH
+      $pfx_password = New-Object -TypeName PSObject
+      $pfx_password | Add-Member -MemberType ScriptProperty -Name "Password" -Value { ("~!@#$%^&*_-+=`|\\(){}[<]:;'>,.?/0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz".tochararray() | Sort-Object { Get-Random })[0..14] -join '' }
+      if (-not (Test-Path HKLM:\\SOFTWARE\\Progress)){
+        New-Item -Path "HKLM:\\SOFTWARE\\Progress\\Authentication" -Force
+        New-ItemProperty  -path "HKLM:\\SOFTWARE\\Progress\\Authentication" -name "PfxPass" -value $pfx_password.Password -PropertyType String
+      }
+    EOH
+  end
+
+  def remove_certificate_from_store
+    powershell_exec! <<~EOH
+      Get-ChildItem -path cert:\\LocalMachine\\My -Recurse -Force  | Where-Object { $_.Subject -Match "#{client_name}" } -ErrorAction Stop | Remove-Item
```

Comment:
do you need a `Catch { return $false }` here? I'm ... unclear

Reply:
added